### PR TITLE
Add standard evaluation setup

### PR DESCRIPTION
**Pull request by yizhongw**
- Add the train/test splits in `splits` folder. 
- Add a new field `Instances Subset for Testing` in json files of the 154 evaluation tasks.
- Add the README for evaluation setup.

### CODE REVIEW

**Comment by danyaljj** [comment]
Thanks! 

I wonder if instead of creating a new field for the 100 subset of instances, we can just say the first 100 instances? If there is a concern regarding reproducibility, we can re-order the instances so that your random evaluation instances appear as the first 100 ones.

**Comment by yizhongw** [comment]
> Thanks!
> 
> I wonder if instead of creating a new field for the 100 subset of instances, we can just say the first 100 instances? If there is a concern regarding reproducibility, we can re-order the instances so that your random evaluation instances appear as the first 100 ones.

Sounds good to me. Let me re-order the instances.

File: splits/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Evaluation Setup of Natural Instruction V2
+
```
Comment (danyaljj) [inline splits/README.md:2]: Suggested addition: 

"Here we describe the evaluation setup used in [this paper](https://arxiv.org/abs/2204.07705) which can be used for reproducing our experiments or extending them."

File: splits/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+To facilitate research on cross-task generalization, we split out tasks into two subsets: one for training and one for testing. Users can use the training tasks for any modeling purpose, while the testing tasks should be used to evaluate the model's ability to solve new tasks by following instruction.
```
Comment (danyaljj) [inline splits/README.md]: Suggested rephrasing: 
"The target setup of the study is a cross-task generalization, i.e., training on a subset of tasks and evaluating on the remaining unseen ones. To do so, we split ... "

File: splits/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## English Track
```
Comment (danyaljj) [inline splits/README.md:7]: These appeared suddenly. I think you need a sentence to connect the splits to "tracks".

File: src/split_tasks.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
+import argparse
+import glob
+import random
+import json 
+import os
+from secrets import choice
+
```
Comment (danyaljj) [inline src/split_tasks.py:14]: Add a few sentences describing what this script does.

**Comment by danyaljj** [comment]
I think we should add a few lies to the testing script (https://github.com/allenai/natural-instructions/blob/master/src/test_all.py) in order to make sure that: 
 - Each task appears in the split  
 - East task in the split actually exists 
 - There is no overlap between test and train task names in the splits files.

**Comment by yizhongw** [comment]
> I think we should add a few lies to the testing script (https://github.com/allenai/natural-instructions/blob/master/src/test_all.py) in order to make sure that:
> 
> * Each task appears in the split
> * East task in the split actually exists
> * There is no overlap between test and train task names in the splits files.

sure, will do!

File: splits/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Evaluation Setup of Natural Instruction V2
```
Comment (danyaljj) [inline splits/README.md:1]: Somewhere you may want to add a reference to your models (https://github.com/yizhongw/Tk-Instruct)

File: src/test_all.py [hunk 6]
```diff
@@ -278,6 +278,21 @@ def natural_keys(text):
...
+    excluded_tasks = [l.strip() for l in open(f"splits/{split_name}/excluded_tasks.txt")]
+
+    # make east task in the split actually exists
+    for task in train_tasks + test_tasks + excluded_tasks:
+        assert task in task_names, f" Task {task} doesn't exist, but it's included in the {split_name} split."
+    # make sure each task appears in the split 
+    for task in task_names:
+        assert task in train_tasks + test_tasks + excluded_tasks, f" Task {task} is missing in the {split_name} split."
+    # make sure there is no overlap between test and train task names in the splits files.
+    assert len(set(train_tasks) & set(test_tasks)) == 0, f" {split_name} split has overlap tasks in the train & test sets."
```
Comment (danyaljj) [inline src/test_all.py:294]: Do the same check with `excluded`?

File: splits/README.md (hunk 5 above)
Comment (danyaljj) [inline splits/README.md:1]: Reference this file from the main readme file?

Comment (danyaljj) [reply splits/README.md:1]: Add a reference to TK-instruct repo to the main readme too!

**Pull request merged**